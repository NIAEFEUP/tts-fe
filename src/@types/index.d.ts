/* General */
export type Student = {
    id: number
    name: string
    email: string
    picture?: string
}

/* Courses */
export type Course = {
    id: number
    course_id: number
    course_unit_id: number
    sigarra_id: number
    course: string
    name: string
    acronym: string
    url: stringid
    course_unit_year: number
    semester: number
    year: number
    schedule_url: string
    ects: number
    last_updated: string
}

export type CheckedCourse = {
    checked: boolean
    info: Course
}

/* Majors */
export type Major = {
    id: number
    name: string
    sigarra_id: number
    faculty: number
    acronym: string
    course_type: string
    year: number
    url: string
    plan_url: string
    last_updated: string
}

/* Schedule */
// export type CourseSchedule = {
//   day: number
//   duration: string
//   start_time: string
//   location: string
//   lesson_type: string
//   teacher_acronym: string
//   course_unit_id: number
//   last_updated: string
//   class_name: string // e.g. 1MIEIC01
//   composed_class_name: string // e.g. COMP752
// }

export type ProfessorInformation = {
    acronym: string
    name: string
}

export type CourseSchedule = {
    day: number
    duration: string
    start_time: string
    location: string
    lesson_type: string
    is_composed: boolean
    course_unit_id: number
    last_updated: string
    class_name: string // e.g. 1MIEIC01
    composed_class_name: string // e.g. COMP752
    // professor_sigarra_id: string
    // professor_acronyms: Array<string>                           // eliminar
    professors_link: string
    professor_information: Array<ProfessorInformation> // new
}

/* Options */
export type CourseOption = {
    shown?: {
        T: boolean
        TP: boolean
    }
    locked: boolean
    course: CheckedCourse
    option: CourseSchedule | null
    schedules: CourseSchedule[]
    teachers: ProfessorInformation[]
    filteredTeachers: ProfessorInformation[]
}

export type Subject = {
    course: Course
    practicalLesson: CourseSchedule[]
    theoreticalLessons: CourseSchedule[]
}

export type Lesson = {
    course: Course
    schedule: CourseSchedule
}

export type MultipleOptions = {
    index: number
    selected: CourseOption[]
    options: CourseOption[][]
}

export type ImportedCourses = {
    [key: string]: string
}

export type ClassExchange = {
    course_unit: string,
    old_class: string,
    new_class: string,
    other_student: string // If the user didn't specify a student, it will be none
}

export type ExchangeCourseUnit = {
    sigla: string,
    name: string,
    class: string,
    code: string
}
