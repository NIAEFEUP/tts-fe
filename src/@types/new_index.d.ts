enum lesson_type {
  T = "T",
  TP = "TP",
  P = "P",
  PL = "PL",
  OT = "OT",
  O = "O",
  E = "E"
}

/* Majors */
export type Major = {
  id: number
  faculty: number
  acronym: string
  name: string
  url: string
  // year: number // remover ??
}

export type CourseInfo = {
  id: number,
  course_unit_year: number,
  course_unit_id: number,
  ects: number,
  acronym: string,
  name: string,
  url: string,
  classes: Array<ClassInfo>
  // last_updated: string, // remover ??
  // semester: number, // remover ??
  // year: number // remover ??
}

export type ClassInfo = {
  // course_unit_id: number, // é mesmo necessário ??
  // composed_name: string,
  id: number,
  name: string,
  filteredTeachers: Array<number>,
  slots: Array<SlotInfo>
}

export type SlotInfo = {
  lesson_type: string,
  day: number,
  start_time: number,
  duration: number,
  location: string,
  professors_link: string,
  professors: Array<ProfessorInfo>,
  // last_updated: string, (is it needed??)
}

export type ProfessorInfo = {
  id: number
  acronym: string
  name: string
}

export type PickedCourses = Array<CourseInfo>

export type MultipleOptions = Array<Option>

export type Option = {
  id: number,
  icon: string,
  name: string,
  course_options: Array<CourseOption>
}

export type CourseOption = {
  course_id: number,
  picked_class_id: number,
  locked: boolean,
  filteredTeachers: Array<number>,
  hide: Array<lesson_type>,
}

// export type ClassDescriptor = {
//   classInfo: ClassInfo
//   courseInfo: CourseInfo
// }

// export type ConflictInfo = {
//     classDescriptor: ClassDescriptor
//     slotInfo: SlotInfo
// }

export type ClassDescriptor = {
  classInfo: ClassInfo
  courseInfo: CourseInfo
  slotInfo?: SlotInfo
}

export type ConflictsInfo = {
  severe: boolean
  classDescriptors: ClassDescriptor[]
}

// |=============================================================|
// |         Ver se nao podemos apagar daqui para baixo:         |
// |=============================================================|

export type Subject = {
  course: Course
  practicalLesson: CourseSchedule[]
  theoreticalLessons: CourseSchedule[]
}

export type Lesson = {
  course: Course
  schedule: CourseSchedule
}

export type ImportedCourses = {
  [key: string]: string
}
