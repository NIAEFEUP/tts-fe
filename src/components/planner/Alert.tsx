import { ReactNode, useMemo } from 'react'
import classNames from 'classnames'
import {
  InformationCircleIcon,
  ExclamationIcon,
  ExclamationCircleIcon,
  CheckCircleIcon,
} from '@heroicons/react/outline'

export enum AlertType {
  info,
  error,
  warning,
  success,
}

type Props = {
  children: ReactNode | ReactNode[]
  type: AlertType
}

const Alert = ({ children, type }: Props) => {
  const animationClass = useMemo(() => {
    if (type === AlertType.error) {
      return 'animate-wiggle'
    } else return ''
  }, [type])

  return (
    <div
      role="alert"
      className={classNames(
        animationClass,
        'flex items-center justify-between rounded border-2 px-2 py-2',
        type === undefined ? 'border-gray-700/20 bg-gray-50 text-gray-700' : '',
        type === AlertType.info ? 'border-sky-700/20 bg-sky-50 text-sky-700' : '',
        type === AlertType.error ? 'border-rose-700/20 bg-rose-50 text-rose-800' : '',
        type === AlertType.warning ? 'border-amber-900/10 bg-orange-50 text-orange-700' : '',
        type === AlertType.success ? 'border-teal-700/30 bg-teal-50 text-teal-600' : ''
      )}
    >
      <div className="inline-flex items-center justify-center space-x-2">
        <span>
          {type === AlertType.info && <InformationCircleIcon className="h-6 w-6 md:h-5 md:w-5" />}
          {type === AlertType.warning && <ExclamationIcon className="h-6 w-6 md:h-5 md:w-5" />}
          {type === AlertType.error && <ExclamationCircleIcon className="h-6 w-6 md:h-5 md:w-5" />}
          {type === AlertType.success && <CheckCircleIcon className="h-6 w-6 md:h-5 md:w-5" />}
        </span>
        <strong className="text-xs font-medium md:text-sm">{children}</strong>
      </div>
    </div>
  )
}

export default Alert
