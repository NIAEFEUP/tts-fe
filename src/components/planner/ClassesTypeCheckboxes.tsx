type Props = {
  classesTHook: [boolean, React.Dispatch<React.SetStateAction<boolean>>]
  classesTPHook: [boolean, React.Dispatch<React.SetStateAction<boolean>>]
}

const ClassesTypeCheckboxes = ({ classesTHook, classesTPHook }: Props) => {
  const [classesT, setClassesT] = classesTHook
  const [classesTP, setClassesTP] = classesTPHook

  return (
    <div className="flex items-center justify-center space-x-4 space-y-0 md:flex-col md:space-y-1 md:space-x-0">
      <div className="flex items-center justify-center space-x-2">
        <input
          type="checkbox"
          id="checkbox-classesT"
          className="checkbox"
          checked={classesT}
          onChange={(event) => setClassesT(event.target.checked)}
        />
        <label className="cursor-pointer text-sm font-medium capitalize tracking-tight" htmlFor="checkbox-classesT">
          <span>Teóricas</span>
        </label>
      </div>
      <div className="flex items-center justify-center space-x-2">
        <input
          type="checkbox"
          id="checkbox-classesTP"
          className="checkbox"
          checked={classesTP}
          onChange={(event) => setClassesTP(event.target.checked)}
        />
        <label className="cursor-pointer text-sm font-medium capitalize tracking-tight" htmlFor="checkbox-classesTP">
          <span>Práticas</span>
        </label>
      </div>
    </div>
  )
}

export default ClassesTypeCheckboxes
