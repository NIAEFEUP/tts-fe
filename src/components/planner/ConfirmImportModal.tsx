import { Dialog, Transition } from '@headlessui/react'
import { XIcon,  CheckIcon} from '@heroicons/react/outline'
import { Fragment, useState } from 'react'

const ConfirmImportModal = (flag : boolean) => {
    const [isOpen, setIsOpen] = useState(flag)
  
    function closeModal() {
      setIsOpen(false)
    }
  
    return (
    <>
    <Transition appear show={isOpen} as={Fragment}>
                <Dialog as="div" className="relative z-10 text-sm" onClose={closeModal}>
                <Transition.Child
                    as={Fragment}
                    enter="ease-out duration-300"
                    enterFrom="opacity-0"
                    enterTo="opacity-100"
                    leave="ease-in duration-200"
                    leaveFrom="opacity-100"
                    leaveTo="opacity-0"
                >
                <div className="fixed inset-0 bg-black bg-opacity-75" />
                </Transition.Child>

                <div className="fixed inset-0 overflow-y-auto">
                    <div className="flex min-h-full items-center justify-center p-4 text-center">
                    <Transition.Child
                        as={Fragment}
                        enter="ease-out duration-300"
                        enterFrom="opacity-0 scale-95"
                        enterTo="opacity-100 scale-100"
                        leave="ease-in duration-200"
                        leaveFrom="opacity-100 scale-100"
                        leaveTo="opacity-0 scale-95"
                    >
                        <Dialog.Panel
                        className="w-full w-1/4 h-50 transform overflow-hidden rounded-2xl bg-white p-6
                        text-left align-middle text-gray-700 shadow-xl transition-all dark:bg-dark dark:text-white"
                        >
                        <div className="flex justify-end">  
                        <button
                            type="button"
                            onClick={closeModal}
                            className="float-right rounded text-rose-700 transition hover:bg-rose-700 hover:text-white"
                            >
                            <XIcon className="h-6 w-6" />
                            </button> 
                        </div>    
                        <div className="flex justify-end">  
                            <div className="mx-auto">              
                                <Dialog.Title as="h1" className="text-2xl rounded-ld font-semibold leading-6 mx-auto">
                                    IMPORTAR  
                                </Dialog.Title>
                            </div>  
                        </div>   
                        <div className="mt-3 flex flex-col text-center">
                            <p>O seguinte processo irá substituir todos horários das unidades curriculares selecionadas.</p>
                        </div>

                        <div className="flex mt-8">
                            <button
                            type="button"
                            className="flex items-center space-x-2 mx-auto rounded bg-tertiary px-3 py-2 text-center text-sm font-medium text-white 
                            transition hover:opacity-80 disabled:cursor-not-allowed disabled:opacity-50"
                            onClick={closeModal}
                            >
                            <span>CONFIRMO</span>
                            <CheckIcon className="h-5 w-5" />
                            </button>
                            
                        </div>
                        </Dialog.Panel>
                    </Transition.Child>
                    </div>
                </div>
                </Dialog>
            </Transition>
        </>
    )

}

export default ConfirmImportModal
