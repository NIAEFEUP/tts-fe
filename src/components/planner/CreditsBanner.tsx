import Alert, { AlertType } from './Alert'
import { useState, useEffect, useMemo } from 'react'
import { CheckedCourse } from '../../@types'

type Props = {
  courses: CheckedCourse[]
}

const CreditsBanner = ({ courses }: Props) => {
  const [alertLevel, setAlertLevel] = useState<AlertType>(AlertType.info)
  const credits = useMemo(() => {
    return courses.length * 6
  }, [courses])

  useEffect(() => {
    if (credits > 42) setAlertLevel(AlertType.error)
    else setAlertLevel(AlertType.info)
  }, [credits])

  return (
    <Alert type={alertLevel}>
      {alertLevel === AlertType.info && (
        <>
          Total número de créditos selecionados: <strong>{credits} ECTS</strong>.
        </>
      )}
      {alertLevel === AlertType.warning && (
        <>
          Créditos selecionados: <strong>{credits} ECTS</strong>. O NIAEFEUP recomenda escolher{' '}
          <strong>36 ou menos</strong>.
        </>
      )}
      {alertLevel === AlertType.error && (
        <>
          Créditos selecionados: <strong>{credits} ECTS</strong>. O limite de créditos da U.Porto num semestre é{' '}
          <strong>42 ECTS</strong>.
        </>
      )}
    </Alert>
  )
}

export default CreditsBanner
