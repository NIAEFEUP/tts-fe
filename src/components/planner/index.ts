import Alert from './Alert'
import Schedule from './Schedule'
import SelectionModal from './SelectionModal'
import ScheduleListbox from './ScheduleListbox'
import ClassesTypeCheckboxes from './ClassesTypeCheckboxes'
import MoreActionsButton from './MoreActionsButton'
import OptionsController from './OptionsController'

export { Schedule, SelectionModal, ScheduleListbox, ClassesTypeCheckboxes, Alert, MoreActionsButton, OptionsController }
