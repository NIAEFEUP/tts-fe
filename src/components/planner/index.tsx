import Alert from './Alert'
import Schedule from './Schedule'
import ExportSchedule from './ExportSchedule'
import SelectionModal from './SelectionModal'
import ScheduleListbox from './ScheduleListbox'
import ClassesTypeCheckboxes from './ClassesTypeCheckboxes'

export { Schedule, SelectionModal, ScheduleListbox, ClassesTypeCheckboxes, ExportSchedule, Alert }
