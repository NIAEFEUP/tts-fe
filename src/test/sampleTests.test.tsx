import { CheckedCourse, Course } from '../@types'
import CreditsBanner from '../components/planner/CreditsBanner'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { AlertType } from '../components/planner/Alert'

describe("Examples of unit testing", () => {
    const getExampleCourse = (ects: number) : CheckedCourse[] => {
        return [{
            checked: true,
            info: {
                id: 1,
                course_id: 1,
                course_unit_id: 1,
                sigarra_id: 1,
                course: "PROG",
                name: "PROG",
                acronym: "PROG",
                url: "1",
                course_unit_year: 1,
                semester: 1,
                year: 1,
                schedule_url: "url",
                ects: ects,
                last_updated: "03/09/2023"
            }
        }]
    }

    test("CreditsBanner should render info with number of ECTS when ECTS < 36", () => {
        render(<CreditsBanner courses={getExampleCourse(20)}></CreditsBanner>)

        const alert = screen.getByRole("alert")
        expect(alert).toBeInTheDocument()
        expect(alert).toHaveTextContent("Total número de créditos selecionados: 20 ECTS.")
    })

    test("CreditsBanner should render error when ECTS > 42", () => {
        render(<CreditsBanner courses={getExampleCourse(43)}></CreditsBanner>)

        const alert = screen.getByRole("alert")
        expect(alert).toBeInTheDocument()
        expect(alert).toHaveTextContent("Créditos selecionados: 43 ECTS. O limite de créditos da U.Porto num semestre, por norma, é 42 ECTS.")
    })
})

describe("Examples of snapshot testing", () => {
    
})  