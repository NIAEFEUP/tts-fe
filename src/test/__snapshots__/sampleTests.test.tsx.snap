// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Examples of snapshot testing Tests Faqs page rendering 1`] = `
<div
  className="container mx-auto w-full max-w-7xl space-y-4 px-4 py-6 md:py-10 md:px-6"
>
  <div>
    <h2
      className="mb-4 text-center text-3xl font-bold capitalize text-primary dark:text-white"
    >
      Perguntas mais frequentes (FAQs)
    </h2>
    <p
      className="w-3/4 text-center mx-auto text-lg"
    >
      No caso de a tua dúvida persistir, contacta-nos pelo email
       
      <a
        className="text-primary transition-all hover:underline hover:opacity-80"
        href="mailto:projetos.niaefeup@gmail.com"
      >
        projetos.niaefeup@gmail.com
      </a>
       
      ou submete uma 
      <span
        className="font-bold underline"
      >
        issue
      </span>
       diretamente no
       
      <a
        className="text-primary transition-all hover:underline hover:opacity-80"
        href="https://github.com/NIAEFEUP/tts-revamp-fe/issues"
      >
        repositório do projeto
      </a>
      .
    </p>
  </div>
  <div
    className="mx-auto flex flex-col items-center justify-center gap-6 pt-14"
    id="planner"
  >
    <div
      className="flex flex-col items-center justify-center"
    >
      <button
        className="relative mb-2 text-center text-3xl font-bold capitalize text-slate-700 transition  before:absolute before:-left-8 hover:opacity-80 hover:before:content-['#'] dark:text-white"
        onClick={[Function]}
      >
        Time Table Selector
      </button>
    </div>
    <div
      className="mx-auto flex w-full flex-col gap-8"
    >
      <div
        className="rounded-2xl bg-white p-3 dark:bg-dark"
        data-headlessui-state=""
      >
        <button
          aria-expanded={false}
          className="group flex w-full items-center justify-between gap-1 rounded-lg bg-slate-200 px-3 py-2 text-sm font-medium tracking-tight text-slate-800 transition hover:bg-slate-700 hover:text-white dark:bg-primary/40 dark:text-white dark:hover:bg-primary/60 lg:px-4 lg:text-base"
          data-headlessui-state=""
          id="headlessui-disclosure-button-:r0:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <span
            className="text-left"
          >
            <span>
              Qual o caso de uso do TTS mais comum?
            </span>
          </span>
          <svg
            aria-hidden="true"
            className="h-5 w-5 min-w-[1.25rem] transition group-hover:text-white lg:h-6 lg:w-6 lg:min-w-[1.5rem]"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m4.5 15.75 7.5-7.5 7.5 7.5"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </button>
      </div>
      <div
        className="rounded-2xl bg-white p-3 dark:bg-dark"
        data-headlessui-state=""
      >
        <button
          aria-expanded={false}
          className="group flex w-full items-center justify-between gap-1 rounded-lg bg-slate-200 px-3 py-2 text-sm font-medium tracking-tight text-slate-800 transition hover:bg-slate-700 hover:text-white dark:bg-primary/40 dark:text-white dark:hover:bg-primary/60 lg:px-4 lg:text-base"
          data-headlessui-state=""
          id="headlessui-disclosure-button-:r1:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <span
            className="text-left"
          >
            <span>
              O que devo fazer se tenho cadeiras em diferentes cursos?
            </span>
          </span>
          <svg
            aria-hidden="true"
            className="h-5 w-5 min-w-[1.25rem] transition group-hover:text-white lg:h-6 lg:w-6 lg:min-w-[1.5rem]"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m4.5 15.75 7.5-7.5 7.5 7.5"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </button>
      </div>
      <div
        className="rounded-2xl bg-white p-3 dark:bg-dark"
        data-headlessui-state=""
      >
        <button
          aria-expanded={false}
          className="group flex w-full items-center justify-between gap-1 rounded-lg bg-slate-200 px-3 py-2 text-sm font-medium tracking-tight text-slate-800 transition hover:bg-slate-700 hover:text-white dark:bg-primary/40 dark:text-white dark:hover:bg-primary/60 lg:px-4 lg:text-base"
          data-headlessui-state=""
          id="headlessui-disclosure-button-:r2:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <span
            className="text-left"
          >
            <span>
              O que devo fazer se não gostar do horário que obtiver?
            </span>
          </span>
          <svg
            aria-hidden="true"
            className="h-5 w-5 min-w-[1.25rem] transition group-hover:text-white lg:h-6 lg:w-6 lg:min-w-[1.5rem]"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m4.5 15.75 7.5-7.5 7.5 7.5"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </button>
      </div>
      <div
        className="rounded-2xl bg-white p-3 dark:bg-dark"
        data-headlessui-state=""
      >
        <button
          aria-expanded={false}
          className="group flex w-full items-center justify-between gap-1 rounded-lg bg-slate-200 px-3 py-2 text-sm font-medium tracking-tight text-slate-800 transition hover:bg-slate-700 hover:text-white dark:bg-primary/40 dark:text-white dark:hover:bg-primary/60 lg:px-4 lg:text-base"
          data-headlessui-state=""
          id="headlessui-disclosure-button-:r3:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <span
            className="text-left"
          >
            <span>
              O TTS deixa-me selecionar as cadeiras que eu quiser?
            </span>
          </span>
          <svg
            aria-hidden="true"
            className="h-5 w-5 min-w-[1.25rem] transition group-hover:text-white lg:h-6 lg:w-6 lg:min-w-[1.5rem]"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m4.5 15.75 7.5-7.5 7.5 7.5"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </button>
      </div>
      <div
        className="rounded-2xl bg-white p-3 dark:bg-dark"
        data-headlessui-state=""
      >
        <button
          aria-expanded={false}
          className="group flex w-full items-center justify-between gap-1 rounded-lg bg-slate-200 px-3 py-2 text-sm font-medium tracking-tight text-slate-800 transition hover:bg-slate-700 hover:text-white dark:bg-primary/40 dark:text-white dark:hover:bg-primary/60 lg:px-4 lg:text-base"
          data-headlessui-state=""
          id="headlessui-disclosure-button-:r4:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <span
            className="text-left"
          >
            <span>
              Para que servem os cadeados ao lado de cada opção de unidade curricular?
            </span>
          </span>
          <svg
            aria-hidden="true"
            className="h-5 w-5 min-w-[1.25rem] transition group-hover:text-white lg:h-6 lg:w-6 lg:min-w-[1.5rem]"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m4.5 15.75 7.5-7.5 7.5 7.5"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
